src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.CrossGen.targets

```diff
@@ -0,0 +1,231 @@
+    </PropertyGroup>
+
+    <NETSdkError Condition="!Exists($(_Crossgen))"
+    ResourceName="UnableToFindResolvedPath"
+    FormatArguments="$(_Crossgen)" />
+
+    <PropertyGroup>
+      <_Crossgen> $([System.IO.Path]::GetFullPath($(_Crossgen)))</_Crossgen>
+    </PropertyGroup>
+
+    <!-- Create the output directory where crossgened assemblies are going to be present -->
+    <ItemGroup>
+      <_RuntimeOptimizedDirectories Include="@(_AssembliestoCrossGen -> '$(_RuntimeOptimizedDir)\%(RecursiveDir)')"/>
+    </ItemGroup>
+    <MakeDir  Directories="@(_RuntimeOptimizedDirectories)"/>
+
+    <!-- CrossGen the assemblies  -->
+
+    <MSBuild Projects="$(MSBuildProjectFullPath)"
+                 Targets="RunCrossGen"
+                 Properties="_CGenExe=$(_Crossgen);
+                             _CGenJit=$(_JitPath);
+                             _CGenInputAssembly=%(_AssembliestoCrossGen.Fullpath);
+                             _CGenOutputAssembly=$([System.IO.Path]::Combine($(_RuntimeOptimizedDir),%(RecursiveDir)%(Filename)%(Extension)));
+                             _CGenPlatformAssembliesPath=$(_RuntimeRefDir)$(PathSeparator)$(_NetCoreRefDir);" />
+
+
+    <!-- Copy by proxy the directory structure to PublishDir-->
+    <ItemGroup>
+      <_ManagedResolvedFileToPublish Include="$(_RuntimeOptimizedDir)\**\*.*" />
+    </ItemGroup>
+  </Target>
+
+  <!--
+    ============================================================
+                                        RunCrossGen
+    Target Encapsulating the crossgen command  
+    ============================================================
+    -->
+  <Target Name="RunCrossGen">
+    <Message Text="$(_CGenExe) -readytorun -in $(_CGenInputAssembly) -out $(_CGenOutputAssembly) -jitpath $(_CGenJit) -platform_assemblies_paths $(_CGenPlatformAssembliesPath)"/>
+
+
+    <Exec
+     Command="$(_CGenExe) -readytorun -in $(_CGenInputAssembly) -out $(_CGenOutputAssembly) -jitpath $(_CGenJit) -platform_assemblies_paths $(_CGenPlatformAssembliesPath)">
+      <Output TaskParameter="ExitCode" PropertyName="ErrorCode"/>
+    </Exec>
+  </Target>
+
+  <!--
+    ============================================================
+                                        _GenerateCrossgenProj
+    Generates the project used to restore netcoreapp correspoding to FX_Version
+    ============================================================
+    -->
+  <Target Name="_GenerateCrossgenProj">
+    <!-- Create a Project to Restore Crossgen -->
+    <PropertyGroup>
+      <_CrossgenProjContent>
+        <![CDATA[
+<Project Sdk="Microsoft.NET.Sdk" ToolsVersion="$(MSBuildToolsVersion)">
+  <PropertyGroup>
+      <TargetFramework>$(_TFM)</TargetFramework>
+      <RuntimeIdentifier>$(RuntimeIdentifier)</RuntimeIdentifier>
+      <DisableImplicitFrameworkReferences Condition="'$(FX_Version)' != ''">true</DisableImplicitFrameworkReferences>  <!-- Preserve this is as netcoreapp is given by the user -->
```

Comment:
>Preserve this is as [](start = 124, length = 19)

A different way to do this would be to set the `<RuntimeFrameworkVersion>$(FX_Version)</RuntimeFrameworkVersion>` property.  Then you wouldn't need both this "disable" property and the PackageReference below.  Setting RuntimeFrameworkVersion property tells the SDK which Microsoft.NETCore.App package to reference implicitly.